cmd/node/node.go

```diff
@@ -952,16 +953,25 @@ func (app *App) LoadOrCreateEdSigner() (*signing.EdSigner, error) {
 		if err := os.MkdirAll(filepath.Dir(filename), 0o700); err != nil {
 			return nil, fmt.Errorf("failed to create directory for identity file: %w", err)
 		}
-		err = os.WriteFile(filename, edSgn.PrivateKey(), 0o600)
+
+		err = os.WriteFile(filename, []byte(hex.EncodeToString(edSgn.PrivateKey())), 0o600)
 		if err != nil {
 			return nil, fmt.Errorf("failed to write identity file: %w", err)
 		}
 
 		log.With().Info("created new identity", edSgn.PublicKey())
 		return edSgn, nil
 	}
+	dst := make([]byte, signing.PrivateKeySize)
+	n, err := hex.Decode(dst, data)
+	if err != nil {
+		return nil, fmt.Errorf("bad hex: %w", err)
```

Comment:
The error could be more descriptive, like
```suggestion
		return nil, fmt.Errorf("decoding private key: %w", err)
```

That would end up as: `"decoding private key: encoding/hex: invalid byte"`, which is IMO more helpful in figuring out what is wrong.